.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,42 @@
+name: "Docker Build CI"
+env:
+  repository: "cloudcustodian"
+  job_metadata: '
+    {
+      "c7n": {
+        "name": "cli",
+        "description": "Cloud Management Rules Engine",
+        "custodian_image_var": "CUSTODIAN_CLI_IMAGE"
+      }
+  }'
+
+
+on: push
+
+jobs:
+
+  Test:
+    strategy:
+      matrix:
+        image:
+          - c7n
+        flavor:
+          - ""
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: ${{ matrix.image }} ${{ matrix.flavor }} build and push
+        uses: ./.github/composites/docker-build-push
+        env:
+          HUB_USER: c7nbuild
+          HUB_TOKEN: ${{ secrets.DOCKER_CI_HUB_TOKEN }}
+          FLAVOR: ${{ matrix.flavor }}
+        with:
+          dockerfile: "docker/${{ fromJSON(env.job_metadata)[matrix.image]['name'] }}${{ matrix.flavor && format('-{0}', matrix.flavor) || matrix.flavor }}"
```

Comment:
it feels like the whole flavor thing could be omitted for docker file name.

afaics atm we don't actually publish the distroless images anymore atm. what's the total build time for the full matrix with distroless vs sans, and what's the actual image size delta for say c7n?

Reply:
im seeing 147MB vs 124MB for default vs distroless in x86, 203MB vs 180MB for ARM64. Total build time is around 20 minutes or so most of the time is spent on the ARM image building, just building x86 is about 5 minutes

Reply:
I think that was part of the rationale on not building it anymore (minimal savings), along with the time for ci infra. we provide docker files for those interested. we shouldn't reintroduce it here as we go to add a multiplier on ci build times for arm.